src/opts.ts

```diff
@@ -67,9 +72,49 @@ export type ToStringable = {
 export type CacheOptions = TargetPath | { target: TargetPath } & Record<string, ToStringable>
 export type CacheMap = Record<SourcePath, CacheOptions>
 
+function getCacheMapFromDockerfile(dockerfilePath: string): CacheMap {
+  const dockerfileContent = fs.readFileSync(dockerfilePath, "utf-8");
+  const dockerfile = DockerfileParser.parse(dockerfileContent);
+
+  const cacheMap: CacheMap = {};
+
+  const runInstructions = dockerfile.getInstructions().filter(i => i.getKeyword() == 'RUN') as Array<ModifiableInstruction>;
+  for (const run of runInstructions) {
+    const currentWorkdir = dockerfile.getAvailableWorkingDirectories(run.getRange().end.line).pop();
+    for (const flag of run.getFlags()) {
+      if (flag.getName() == 'mount' && flag.getOption('type')?.getValue() == 'cache') {
+        const cacheTarget = flag.getOption('target')?.getValue();
+        if (cacheTarget == null) {
+          throw new Error('cache mount must define target: ' + flag.toString() + ' in ' + run.toString());
+        }
+        if (!cacheTarget.startsWith('/') && !currentWorkdir) {
+          throw new Error('cache mount target must be absolute path or WORKDIR must be set before: ' + flag.toString() + ' in ' + run.toString());
+        }
+
+        // Extract the `id` flag which defaults to `target` when `id` is not set
+        // https://docs.docker.com/reference/dockerfile/#run---mounttypecache
+        const id = flag.getOption('id')?.getValue() || cacheTarget;
+        const target = cacheTarget?.startsWith('/') ? cacheTarget : currentWorkdir + cacheTarget;
```

Comment:
I don't believe the `target` we specify here needs to be precisely what is used by the user's Dockerfile as target as used by extraction/injestion could be anything as long as it isn't `/var/dance-cache`. What matters in the context of this action is the `id` (which does fall back on the `target`).

I just mention this as you could skip the `currentWorkdir` processing as it should be fine to hardcode this as `/var/cache-target`.